TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/CapturedTrafficToHttpTransactionAccumulator.java

```diff
@@ -90,22 +101,41 @@ private static String summarizeTrafficStream(TrafficStream ts) {
 
     public void accept(TrafficStream yetToBeSequencedTrafficStream) {
         log.trace("Got trafficStream: " + summarizeTrafficStream(yetToBeSequencedTrafficStream));
-        var accum = liveStreams.getOrCreateWithoutExpiration(
-                yetToBeSequencedTrafficStream.getNodeId(),
-                yetToBeSequencedTrafficStream.getConnectionId());
+        var partitionId = yetToBeSequencedTrafficStream.getNodeId();
+        var connectionId = yetToBeSequencedTrafficStream.getConnectionId();
+        var accum = liveStreams.getOrCreateWithoutExpiration(partitionId, connectionId);
+        var terminated = new AtomicBoolean(false);
         accum.sequenceTrafficStream(yetToBeSequencedTrafficStream,
                 ts -> ts.getSubStreamList().stream()
-                        .forEach(o -> addObservationToAccumulation(ts.getNodeId(), accum, o)));
+                        .map(o -> {
+                            if (terminated.get()) {
+                                log.error("Got a traffic observation AFTER a Close observation for the stream " +  ts);
+                                return true;
+                            }
+                            var didTerminate = CONNECTION_STATUS.CLOSED ==
+                                    addObservationToAccumulation(ts.getNodeId(), accum, o);
+                            terminated.set(didTerminate);
+                            return didTerminate;
+                        })
+                        .takeWhile(b->!b)
+                        .forEach(b->{}));
```

Comment:
Curious what this forEach line is doing?

Reply:
Okay this makes more sense now that we are juggling a bit of processing this stream as well as reserving the ability to terminate early if the connection is closed. Part of me thinks a simple while loop instead of a stream would have been cleaner here, but its not a blocker for me